index.src.html

```diff
@@ -656,6 +656,31 @@ <h2 id="integration-with-html">Integration with HTML</h2>
 </section>
 
 <section>
+  <h2 id="integration-with-css">Integration with CSS</h2>
+
+  The CSS Standard does not specify how it fetches resources referenced from
+  stylesheets. However, implementations use the URL a given stylesheet was
```

Comment:
Yes, and the spec that definition is in, plus other specs that might manipulate document URLs, defines what that URL is for each Document object.   For example, the very next paragraph says:

> Unless stated otherwise ... URL is "about:blank"

So in practice, an implementor has a field called "URL" on document objects.  When a document is created, set the field to "about:blank".  When some spec says to change the document's URL, change it accordingly.  For example, https://html.spec.whatwg.org/multipage/browsers.html#set-the-document%27s-address does this during navigation.  https://html.spec.whatwg.org/multipage/browsers.html#dom-history-pushstate step 10 does this when pushState or replaceState is called.  That sort of thing.

Then when some spec says to use the document's URL, use the value of that field.  Everything is clearly defined, no ambiguity.